SD-K9/src/app/components/building-info/building-info.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  animate,
+  transition
+} from '@angular/animations';
+import { UnsubscribeCallback } from 'src/app/interfaces/unsubscribe-callback';
+import { BuildingInfoStore } from 'src/app/providers/state-stores/building-info-store.service';
+import { MapModeStore } from '../../providers/state-stores/map-mode-store.service';
+import { ViewMode } from 'src/app/models/view-mode.enum.model';
+
+import * as campusData from '../../../local-configs/campus.json';
+
+@Component({
+  selector: 'app-building-info',
+  templateUrl: './building-info.component.html',
+  styleUrls: ['./building-info.component.scss'],
+  animations: [
+    // the fade-in/fade-out animation.
+    trigger('cardAnimation', [
+      state('hidden', style({
+        opacity: 0,
+        transform: "translate(0px,100px)",
+        height:"0px"
+      })),
+      state('shown', style({
+        opacity: 1,
+        transform: "translate(0px,0px)",
+        height:"auto"
+      })),
+      transition('hidden => shown', [
+        animate('0.2s')
+      ]),
+      transition('shown => hidden', [
+        animate('0.2s')
+      ]),
+    ])
+  ]
+})
+export class BuildingInfoComponent implements OnInit, OnDestroy {
+  currentBuilding: string;
+  currentCampus: string;
+  buildingSlug: string = '';
+  currentBuildingInfo: string;
+  buildingInfoCardIsShown: boolean;
+
+  private _unsubscribe: UnsubscribeCallback;
+  private _campusConfig: any;
+
+  constructor(
+    private _buildingInfoStore: BuildingInfoStore,
+    private _mapModeStore: MapModeStore
+  ) {
+    this._unsubscribe = this._buildingInfoStore.subscribe(() => {
```

Comment:
Unsubscribe is calling subscribe? I'm a little confused as to how this works!

Reply:
The Observable returned from this._buildingInfoStore.subscribe() is stored in the variable _unsubscribe so that when the component is destroyed, we can later unsubscribe using that variable.

Reply:
Okay that makes sense!